configure.ac

```diff
@@ -216,11 +216,48 @@ AC_SUBST([ocaml_bindir])
 AC_SUBST([ocaml_libdir])
 AC_SUBST([QS])
 AC_SUBST([ar_supports_response_files])
+AC_SUBST([vpath])
 
 ## Generated files
 
+AC_DEFUN([AC_CONFIG_LINK_SELF], [AC_CONFIG_LINKS([$1:$1])])
+
 AC_CONFIG_FILES([Makefile.build_config])
 AC_CONFIG_FILES([Makefile.config])
+AS_IF([test x"$srcdir" != "x."],
+  [vpath='VPATH := $(SRCDIR)'
+   AC_CONFIG_LINK_SELF([Makefile])
+   AC_CONFIG_LINK_SELF([Makefile.common])
+   AC_CONFIG_LINK_SELF([Makefile.best_binaries])
+   AC_CONFIG_LINK_SELF([Makefile.config_if_required])
+   AC_CONFIG_LINK_SELF([stdlib/Makefile])
+   AC_CONFIG_LINK_SELF([otherlibs/Makefile])
+   AC_CONFIG_LINK_SELF([otherlibs/Makefile.otherlibs.common])
+   AC_CONFIG_LINK_SELF([otherlibs/str/Makefile])
+   AC_CONFIG_LINK_SELF([otherlibs/unix/Makefile])
+   AC_CONFIG_LINK_SELF([otherlibs/runtime_events/Makefile])
+   AC_CONFIG_LINK_SELF([otherlibs/systhreads/Makefile])
+   AC_CONFIG_LINK_SELF([otherlibs/dynlink/Makefile])
+   AC_CONFIG_LINK_SELF([api_docgen/Makefile])
+   AC_CONFIG_LINK_SELF([api_docgen/Makefile.common])
+   AC_CONFIG_LINK_SELF([api_docgen/odoc/Makefile])
+   AC_CONFIG_LINK_SELF([api_docgen/ocamldoc/Makefile])
+   AC_CONFIG_LINK_SELF([man/Makefile])
+   AC_CONFIG_LINK_SELF([debugger/Makefile])
+   AC_CONFIG_LINK_SELF([ocamldoc/Makefile])
+   AC_CONFIG_LINK_SELF([ocamldoc/Makefile.best_ocamldoc])
+   AC_CONFIG_LINK_SELF([ocamltest/Makefile])
+   AC_CONFIG_LINK_SELF([testsuite/Makefile])
+   AC_CONFIG_LINK_SELF([testsuite/lib/Makefile])
+   AC_CONFIG_LINK_SELF([testsuite/tools/Makefile])
+   AC_CONFIG_LINK_SELF([manual/Makefile])
+   AC_CONFIG_LINK_SELF([manual/tools/Makefile])
+   AC_CONFIG_LINK_SELF([manual/tests/Makefile])
+   AC_CONFIG_LINK_SELF([manual/src/Makefile])
+   AC_CONFIG_LINK_SELF([manual/src/cmds/Makefile])
+   AC_CONFIG_LINK_SELF([manual/src/library/Makefile])
+   AC_CONFIG_LINK_SELF([manual/src/refman/Makefile])
+   AC_CONFIG_LINK_SELF([manual/src/tutorials/Makefile])])
```

Comment:
It is not obvious why these AC_CONFIG_LINK_SELF lines are needed, a comment could help.